plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/FilesTable.java

```diff
@@ -174,45 +180,85 @@ public CloseableIterator<List<Object>> iterator()
             addCloseable(planFilesIterator);
 
             return new CloseableIterator<>() {
+                private CloseableIterator<StructLike> currentIterator = CloseableIterator.empty();
+
                 @Override
                 public boolean hasNext()
                 {
-                    return !closed && planFilesIterator.hasNext();
+                    updateCurrentIterator();
+                    return !closed && currentIterator.hasNext();
                 }
 
                 @Override
                 public List<Object> next()
                 {
-                    return getRecord(planFilesIterator.next().file());
+                    updateCurrentIterator();
+
+                    StructLike dataTaskFile = currentIterator.next();
+
+                    StructLike innerContentFile = null;
+
+                    // the StructWithReadableMetrics is a private class in iceberg-core
+                    // so we need to use reflection to access it
+                    try {
+                        Class<?> clazz = dataTaskFile.getClass();
+                        Field field = clazz.getDeclaredField("struct");
+                        field.setAccessible(true);
+                        innerContentFile = (StructLike) field.get(dataTaskFile);
+                    }
```

Comment:
Is it possible to avoid reflection if we manage a map (name -> position) like `SnapshotsTable`?